plugins/plugin-gdb/che-plugin-gdb-ide/src/main/java/org/eclipse/che/plugin/gdb/ide/configuration/GdbConfigurationPagePresenter.java

```diff
@@ -116,24 +119,38 @@ public void go(AcceptsOneWidget container) {
     }
 
     private void setHostsList() {
-        machineServiceClient.getMachines(appContext.getWorkspaceId()).then(new Operation<List<MachineDto>>() {
-            @Override
-            public void apply(List<MachineDto> machines) throws OperationException {
-                @SuppressWarnings("unchecked")
-                Promise<RecipeDescriptor>[] recipePromises = (Promise<RecipeDescriptor>[])new Promise[machines.size()];
-
-                for (int i = 0; i < machines.size(); i++) {
-                    String location = machines.get(i).getConfig().getSource().getLocation();
-                    String recipeId = getRecipeId(location);
-                    recipePromises[i] = recipeServiceClient.getRecipe(recipeId);
-                }
+        List<Machine> machines = getMachines();
+        if (machines.isEmpty()) {
+            return;
+        }
+
+        @SuppressWarnings("unchecked")
+        Promise<RecipeDescriptor>[] recipePromises = (Promise<RecipeDescriptor>[])new Promise[machines.size()];
+        for (int i = 0; i < machines.size(); i++) {
+            String location = machines.get(i).getConfig().getSource().getLocation();
+            String recipeId = getRecipeId(location);
+            recipePromises[i] = recipeServiceClient.getRecipe(recipeId);
+        }
+        setHostsList(recipePromises, machines);
+    }
+
+    private List<Machine> getMachines() {
+        List<Machine> machines = new ArrayList<>();
+        Workspace workspace = appContext.getWorkspace();
+        if (workspace == null || workspace.getRuntime() == null) {
+            return machines;
```

Comment:
If I understand right, we can use `Collections.emptyList()` here and further in similar cases. There is a short note on the topic here: http://stackoverflow.com/questions/5552258/collections-emptylist-vs-new-instance